drivers/s3.go

```diff
@@ -38,47 +38,51 @@ type s3OS struct {
 	awsAccessKeyID     string
 	awsSecretAccessKey string
 	s3svc              *s3.S3
+	gsSigner           *gsSigner
+	isGoogle           bool
 }
 
 type s3Session struct {
-	host           string
-	key            string
-	policy         string
-	signature      string
-	xAmzCredential string
-	xAmzDate       string
-	driver         *s3OS
+	host       string
+	key        string
+	policy     string
+	signature  string
+	credential string
+	xAmzDate   string
+	isGoogle   bool
 }
 
 // S3BUCKET s3 bucket owned by this node
 var S3BUCKET string
 
-// S3REGION region of s3 bucket owned by this node
-var S3REGION string
-
-func s3Host(bucket string) string {
+func s3Host(bucket string, isGoogle bool) string {
+	if isGoogle {
+		return fmt.Sprintf("https://%s.storage.googleapis.com", bucket)
+	}
 	return fmt.Sprintf("https://%s.s3.amazonaws.com", bucket)
 }
 
 // IsOwnStorageS3 returns true if turi points to S3 bucket owned by this node
 func IsOwnStorageS3(uri string) bool {
-	return strings.HasPrefix(uri, s3Host(S3BUCKET))
+	return strings.HasPrefix(uri, s3Host(S3BUCKET, true)) ||
+		strings.HasPrefix(uri, s3Host(S3BUCKET, false))
 }
 
 func newS3Session(info *net.S3OSInfo) OSSession {
 	return &s3Session{
-		host:           info.Host,
-		key:            info.Key,
-		policy:         info.Policy,
-		signature:      info.Signature,
-		xAmzDate:       info.XAmzDate,
-		xAmzCredential: info.XAmzCredential,
+		host:       info.Host,
+		key:        info.Key,
+		policy:     info.Policy,
+		signature:  info.Signature,
+		xAmzDate:   info.XAmzDate,
+		credential: info.Credential,
+		isGoogle:   info.IsGoogle,
```

Comment:
Adding Google to the `StorageType` field would allow us to use a separate constructor for Google and avoid setting this field for S3.

```golang
func newGSSession(info *net.S3OSInfo) OSSession {
   sess := newS3Session(info).(*s3Session)
   sess.isGoogle = true
   return sess
}
```